### PR TITLE
Fix/permalinks

### PR DESCRIPTION
**Pull request by dlouise64**
- Bit of refactoring done.
- Now using redux.
- Permalinks working - although requesting by FSD_ID returns multiple results for some services.
- Fixed drop down mishap - @Br3nda there is no lat long for these addresses so have not continued on this part, will start in a new feature.
- Added CSP
- Filters menu now responsive, uses select menu in mobile

TODO:
- Localisation

- [x] has automated tests
- [ ] at least one a reviewer ran the code
- [x] tested in small screen
- [ ] i18n
- [x] tested in screen reader/contrast

### CODE REVIEW

**Comment by Br3nda** [comment]
If we merge #72, hound  will be quieter

File: src/__tests__/Redux.test.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import * as actions from '../actions/index';
+
+describe('actions', () => {
```
Comment (Br3nda) [inline src/__tests__/Redux.test.js]: so happy we have tests!

File: src/actions/index.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import axios from 'axios';
+
+const resourceId = '35de6bf8-b254-4025-89f5-da9eb6adf9a0';
+
+export function loadFilters(){
+  let sql = encodeURI(`SELECT "LEVEL_1_CATEGORY" as name, COUNT(*) as num FROM "${resourceId}" GROUP BY name ORDER BY name`);
+  let url = `https://catalogue.data.govt.nz/api/3/action/datastore_search_sql?sql=${sql}`;
```
Comment (Br3nda) [inline src/actions/index.js]: We should move the URL into config. or at least have it in only one place.

File: src/actions/index.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+const fields = ()=> {
+  return 'LEVEL_1_CATEGORY,FSD_ID,LONGITUDE,LATITUDE,PROVIDER_NAME,PUBLISHED_CONTACT_EMAIL_1,PUBLISHED_PHONE_1,PROVIDER_CONTACT_AVAILABILITY,ORGANISATION_PURPOSE,PHYSICAL_ADDRESS,SERVICE_NAME,SERVICE_DETAIL,DELIVERY_METHODS,COST_TYPE,SERVICE_REFERRALS';
+};
+
+export function loadResults(category, keyword) {
+  
+  let url = `https://catalogue.data.govt.nz/api/3/action/datastore_search?resource_id=${resourceId}&fields=${fields()}`;
+  var query;
+  if (keyword.length > 2) {
```
Comment (Br3nda) [inline src/actions/index.js]: i'm thinking this `if` could refactor down a bit more to read like english
:elephant:

Comment (dlouise64) [reply src/actions/index.js]: @Br3nda good spotting, that was definitely a code smell. See what I've done now, I moved the conditionals into their own functions.

File: src/actions/index.js [hunk 4]
```diff
@@ -1,10 +1,11 @@
 import axios from 'axios';
+
 const RESOURCE_ID = process.env.REACT_APP_API_RESOURCE_ID;
 const API_PATH = `${process.env.REACT_APP_API_PATH}`;
 
 let filters = category => category ? `{"LEVEL_1_CATEGORY":"${category}"}` : '';
-let q = keyword => keyword.length > 2 ? `${keyword}` : '';
-const fields = () => 'LEVEL_1_CATEGORY,FSD_ID,LONGITUDE,LATITUDE,PROVIDER_NAME,PUBLISHED_CONTACT_EMAIL_1,PUBLISHED_PHONE_1,PROVIDER_CONTACT_AVAILABILITY,ORGANISATION_PURPOSE,PHYSICAL_ADDRESS,SERVICE_NAME,SERVICE_DETAIL,DELIVERY_METHODS,COST_TYPE,SERVICE_REFERRALS';
+let q = keyword => keyword.length > 2 ? keyword : '';
+const FIELDS = 'LEVEL_1_CATEGORY,FSD_ID,LONGITUDE,LATITUDE,PROVIDER_NAME,PUBLISHED_CONTACT_EMAIL_1,PUBLISHED_PHONE_1,PROVIDER_CONTACT_AVAILABILITY,ORGANISATION_PURPOSE,PHYSICAL_ADDRESS,SERVICE_NAME,SERVICE_DETAIL,DELIVERY_METHODS,COST_TYPE,SERVICE_REFERRALS';
```
Comment (dlouise64) [inline src/actions/index.js]: @Br3nda I changed this to a string instead of a function.

**Pull request merged**